File: src/main/java/io/github/thehrz/snowcraft/object/machines/AbstractBlockCompresMachine.java
package io.github.thehrz.snowcraft.object.machines;

import me.mrCookieSlime.Slimefun.Lists.RecipeType;
import me.mrCookieSlime.Slimefun.Objects.Category;
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer;
import org.bukkit.Material;
import org.bukkit.inventory.ItemStack;

/**
 * @author Thehrz
 */
public abstract class AbstractBlockCompresMachine extends AContainer {
    private final String id;

    public AbstractBlockCompresMachine(Category category, ItemStack item, String id, RecipeType recipeType, ItemStack[] recipe) {
        super(category, item, id, recipeType, recipe);
        this.id = id;
    }


    @Override
    public String getInventoryTitle() {
        return "§a方块压缩机";
    }

    @Override
    public ItemStack getProgressBar() {
        return new ItemStack(Material.PISTON_BASE);
    }

    @Override
    public void registerDefaultRecipes() {
        this.registerRecipe(5, new ItemStack[]{new ItemStack(Material.GRAVEL)}, new ItemStack[]{new ItemStack(Material.NETHERRACK, 2)});
        this.registerRecipe(3, new ItemStack[]{new ItemStack(Material.PACKED_ICE)}, new ItemStack[]{new ItemStack(Material.SNOW_BLOCK)});
    }

    @Override
    public String getMachineIdentifier() {
        return this.id;
    }

}

File: src/main/kotlin/io/github/thehrz/snowcraft/object/machines/AbstractBlockCompresMachine.kt
package io.github.thehrz.snowcraft.`object`.machines

import me.mrCookieSlime.Slimefun.Lists.RecipeType
import me.mrCookieSlime.Slimefun.Objects.Category
import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.AContainer
import org.bukkit.Material
import org.bukkit.inventory.ItemStack
import taboolib.common.platform.function.console
import taboolib.module.lang.asLangText

/**
 * @author Thehrz
 */
abstract class AbstractBlockCompresMachine(
    category: Category?,
    item: ItemStack?,
    id: String,
    recipeType: RecipeType?,
    recipe: Array<ItemStack?>?
) : AContainer(category, item, id, recipeType, recipe) {
    override fun registerDefaultRecipes() {
        registerRecipe(5, arrayOf(ItemStack(Material.GRAVEL)), arrayOf(ItemStack(Material.NETHERRACK, 2)))
        registerRecipe(3, arrayOf(ItemStack(Material.PACKED_ICE)), arrayOf(ItemStack(Material.SNOW_BLOCK)))
    }

    override fun getInventoryTitle() = console().asLangText("block-compres-machine")

    override fun getProgressBar() = ItemStack(Material.PISTON_BASE)

    override fun getMachineIdentifier() = id!!
}